### PR TITLE
Feature/add integration events into what backend

### PR DESCRIPTION
**Pull request by SerjKocherga**
1.  Integrated event-base communication between microservices through cloudampq.
2.  Added subscription to EmailRenderService for AccountApproveEvent.
3.  Added publishing of created email body to EmailSenderService.
4.  Added NotificationService.
5.  EmailCredentialsSenderService was removed from WHAT project.

### CODE REVIEW

**Review by SnipeFrost — APPROVED** [review]
good

File: CharlieBackend.Business/Services/StudentService.cs [hunk 1]
```diff
@@ -58,8 +57,8 @@ public async Task<Result<StudentDto>> CreateStudentAsync(long accountId)
 
                     await _unitOfWork.CommitAsync();
 
-                    _bus.PubSub.Publish(new AccountApprovedEvent(account.Email,
-                                        account.FirstName, account.LastName, account.Role));
+                    await _bus.SendReceive.SendAsync("EmailRenderService", new AccountApprovedEvent(account.Email,
+                                       account.FirstName, account.LastName, account.Role));
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/StudentService.cs:61]: extract this into INotificationService.AccountApproved(Account account)
and remove IBus from this class

File: CharlieBackend.Business/Services/SecretaryService.cs [hunk 2]
```diff
@@ -56,8 +55,8 @@ public async Task<Result<SecretaryDto>> CreateSecretaryAsync(long accountId)
 
                     await _unitOfWork.CommitAsync();
 
-                    _bus.PubSub.Publish(new AccountApprovedEvent(account.Email,
-                                        account.FirstName, account.LastName, account.Role));
+                    await _bus.SendReceive.SendAsync("EmailRenderService", new AccountApprovedEvent(account.Email,
+                                       account.FirstName, account.LastName, account.Role));
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/SecretaryService.cs:59]: extract this into INotificationService.AccountApproved(Account account)
and remove IBus from this class

File: CharlieBackend.Business/Services/MentorService.cs [hunk 3]
```diff
@@ -58,8 +56,8 @@ public async Task<Result<MentorDto>> CreateMentorAsync(long accountId)
 
                     await _unitOfWork.CommitAsync();
 
-                    _bus.PubSub.Publish(new AccountApprovedEvent(account.Email,
-                                        account.FirstName, account.LastName, account.Role));
+                    await _bus.SendReceive.SendAsync("EmailRenderService", new AccountApprovedEvent(account.Email,
+                                       account.FirstName, account.LastName, account.Role));
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/MentorService.cs:60]: extract this into INotificationService.AccountApproved(Account account)
and remove IBus from this class

File: WhatBackend.EmailRenderService/WhatBackend.EmailRenderService/WhatBackend.EmailRenderService.sln [hunk 4]
```diff
@@ -0,0 +1,31 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (whereismyroot) [inline WhatBackend.EmailRenderService/WhatBackend.EmailRenderService/WhatBackend.EmailRenderService.sln:2]: Please move sln into WhatBackend\WhatBackend.EmailRenderService.sln. it helps to understand which solutions do we have, when we're in the root folder of our projects

File: WhatBackend.EmailRenderService/WhatBackend.EmailRenderService/IntegrationEvents/EventHandling/AccountApprovedConsumer.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+using CharlieBackend.Core.IntegrationEvents.Events;
+
+namespace WhatBackend.EmailRenderService.IntegrationEvents.EventHandling
+{
+    public class AccountApprovedConsumer : IConsumeAsync<AccountApprovedEvent>
+    {
+        #region
+        private readonly ILogger<AccountApprovedConsumer> _logger;
+        private readonly IBus _bus;
+        #endregion
```
Comment (whereismyroot) [inline WhatBackend.EmailRenderService/WhatBackend.EmailRenderService/IntegrationEvents/EventHandling/AccountApprovedConsumer.cs:17]: remove region

File: CharlieBackend.Core/Entities/EmailData.cs [hunk 6]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace CharlieBackend.Core.Entities
+{
+    public class EmailData
```
Comment (whereismyroot) [inline CharlieBackend.Core/Entities/EmailData.cs:5]: This souldnt be located in entities since it's not in our schema. Place this into dto folder

**Review by whereismyroot — CHANGES_REQUESTED** [review]
Need some changes

**Review by whereismyroot — APPROVED** [review]

**Pull request merged**